backend/src/main/java/com/example/cms/security/SecurityConfig.java

```diff
@@ -144,4 +144,12 @@ public SecurityFilterChain notSecuredFilterChain(HttpSecurity http) throws Excep
 
         return http.build();
     }
+
+    @Override
+    @Bean(name = "asyncExecutor")
+    public Executor getAsyncExecutor() {

```

Comment:
To sprawia, że w metodach oznaczonych jako `@Async` mamy nadal dostęp do tego samego `SecurityContext` co miała funkcja, która wywołała methodę `@Async`. W ten sposób, możemy mieć metodę oznaczoną jako `@Secured` i `@Async` bez martwienia się o wyjątki, że w kontekście nie ma użytkownika. 

Skopiowane z jakiejś dokumentacji springa.